### PR TITLE
fix(checkbox): Change default id to be unique per instance

### PR DESCRIPTION
**Pull request by mannycarrera4**
## Summary
Fixes: https://github.com/Workday/canvas-kit/issues/158

### CODE REVIEW

File: modules/checkbox/react/lib/Checkbox.tsx [hunk 1]
```diff
@@ -251,7 +250,7 @@ export default class Checkbox extends React.Component<CheckboxProps> {
           <CheckboxInput
             checked={checked}
             disabled={disabled}
-            id={id}
+            id={id || this.id}
```
Comment (mannycarrera4) [inline modules/checkbox/react/lib/Checkbox.tsx]: Should we add a story with multiple checboxes? @NicholasBoll

Comment (anicholls) [reply modules/checkbox/react/lib/Checkbox.tsx]: Nope, but we should add a test for it @mannycarrera4

File: modules/checkbox/react/lib/Checkbox.tsx [hunk 2]
```diff
@@ -228,9 +228,8 @@ export default class Checkbox extends React.Component<CheckboxProps> {
   public static defaultProps = {
     checked: false,
     label: '',
-    id: uuid(),
   };
-
+  private id = uuid();
```
Comment (anicholls) [inline modules/checkbox/react/lib/Checkbox.tsx]: Nit: can you put new lines on either side of this? We generally put a space between all class members/functions

**Comment by lychyi** [comment]
@mannycarrera4 this one's been kind of stale for a while, are we waiting on something for this?

**Comment by stephanerangaya** [comment]
> @mannycarrera4 this one's been kind of stale for a while, are we waiting on something for this?

Looks like the only thing that was blocking this was writing a test, which @mannycarrera4 did. I think it's good to be merged.

File: modules/checkbox/react/lib/Checkbox.tsx [hunk 3]
```diff
@@ -251,7 +252,7 @@ export default class Checkbox extends React.Component<CheckboxProps> {
           <CheckboxInput
             checked={checked}
             disabled={disabled}
-            id={id}
+            id={id || this.id}
```
Comment (lychyi) [inline modules/checkbox/react/lib/Checkbox.tsx]: Nit: when we destructure `id` out of `props` (L240) we should just assign `this.id` as a default value. This way if this logic ever needs to change you're not changing it in two places (here and L270).

**Pull request merged**